harness/determined/common/context.py

```diff
@@ -109,19 +109,25 @@ def add_item(f: bindings.v1File) -> None:
     # however, the Python documentation claims a warning that rglob may be
     # inefficient on large directory trees, so we use the older os.walk().
     for parent, dirs, files in os.walk(str(root_path)):
+        keep_dirs = []
```

Comment:
sorry, got distracted while reviewing.

This is like the tiniest nit, but I think it's easier to think about code when things are happening explicitly rather than implicitly.  Maybe convert this to `skip_dirs`, add to it right before each `continue`, and call `for directory in skip_dirs: dirs.remove(directory)`.

Same effect, but a little easier to think about.

I won't be offended if you totally ignore this comment either, ha.

Can you just do `dirs.remove(directory)` instead of this keep_dirs pattern?  Or does that break the iteration?